src/Utilities.cpp

```diff
@@ -547,57 +547,80 @@ double getSemiMinorRadius(json isd, csm::WarningList *list) {
   return radius;
 }
 
-std::vector<double> getTransverseDistortionX(json isd, csm::WarningList *list) {
-  std::vector<double> coefficients;
-  try {
-    coefficients = isd.at("optical_distortion").at("transverse").at("x").get<std::vector<double>>();
-    coefficients.resize(10, 0.0);
+DistortionType getDistortionModel(json isd, csm::WarningList *list) {
+  json distoriton_subset = isd.at("optical_distortion");
+
+  json::iterator it = distoriton_subset.begin();
+
+  std::string distortion = (std::string)it.key();
+
+  if (distortion.compare("transverse") == 0) {
+    return DistortionType::TRANSVERSE;
   }
-  catch (...) {
-    if (list) {
-      list->push_back(
-        csm::Warning(
-          csm::Warning::DATA_NOT_AVAILABLE,
-          "Could not parse the transverse distortion model X coefficients.",
-          "Utilities::getTransverseDistortionX()"));
-    }
+  else if (distortion.compare("radial") == 0) {
+    return DistortionType::RADIAL;
```

Comment:
You could use [find](https://nlohmann.github.io/json/classnlohmann_1_1basic__json_a89eb3928f57903677051c80534be9cb1.html#a89eb3928f57903677051c80534be9cb1) here instead of iterating and checking each object